Frends.IMAP.ReadEmail/Frends.IMAP.ReadEmail/IMAP.cs

```diff
@@ -80,6 +88,61 @@ public static List<EmailMessageResult> ReadEmail([PropertyTab] IMAPSettings sett
                     }
                 }
 
+                // save attachments?
+                if (settings.SaveAttachments && !string.IsNullOrEmpty(settings.ArchiveDirectory))
+                {
+                    //check existence of directory
+                    bool _exist = Directory.Exists(settings.ArchiveDirectory);
+
+                    //if not existing and set to create new, then proceed
+                    if(!_exist && options.CreateDirectoryIfNotFound)
+                        Directory.CreateDirectory(settings.ArchiveDirectory);
+                    
+                    //check again, creation might not have worked
+                    _exist = Directory.Exists(settings.ArchiveDirectory);
+
+                    //if directory exists, copy attachments there
+                    if(_exist){
+                        
+                        for (int i = 0; i < attachments.Count; i++)
+                        {
+                            if(attachments[i].Count() > 0)
+                            {
+                                string directoryName = settings.ArchiveDirectory+"/"+result[i].Date.ToString("yyyy-MM-dd HH:mm:ss");
+                                Directory.CreateDirectory(directoryName);
+
+                                foreach (var attachment in attachments[i]) {
+                                    string _directory = directoryName+"/";
+                                    
+                                    if (attachment is MessagePart) {
+                                        var fileName = attachment.ContentDisposition?.FileName;
+                                        var rfc822 = (MessagePart) attachment;
+                                        _directory += fileName;
+
+                                        if (string.IsNullOrEmpty (fileName))
+                                            fileName = "attached-message.eml";
+
+                                        using (var stream = File.Create (_directory))
```

Comment:
Here you still have the _directory var but it actually already contains the file path. Please avoid reusing vars like this as it will lead to confusion for next person doing some new improvements.